syntax = "proto3";

package skaiproto.vehicle;

import "skaiproto/SkaimotProtoMsg.proto";


message VehicleProtoMsg {
    // define frame type for vehicle
    message CameraFrame {
        uint64 timestamp = 1; // epoch time double * 1e9 turned to uint64
        uint64 camera_id = 2; // mac address

        // vehicle type
        message Vehicle {
            skaiproto.skaimot.TLBR_Box box = 1; // top left bottom right floats from 0 to 1
            repeated string object_tags = 2;     
        }

        repeated Vehicle vehicles = 3; // list of vehicle objects        
    }

    // fields here
    repeated CameraFrame camera_frames = 1;
}

message VehicleSpotMonitorProtoMsg {
    uint64 timestamp = 1;
}